cmd/zrepl/zrepl.c

```diff
@@ -285,13 +295,19 @@ uzfs_zvol_worker(void *arg)
 	metadata_desc_t	**metadata_desc;
 	int		rc = 0;
 	int 		write = 0;
+	boolean_t	rebuild;
 
 	zio_cmd = (zvol_io_cmd_t *)arg;
 	hdr = &zio_cmd->hdr;
 	zinfo = zio_cmd->zv;
 	zvol_state = zinfo->zv;
-	/* If zvol hasn't passed rebuild phase we need the metadata */
-	if (ZVOL_IS_REBUILDED(zvol_state)) {
+	rebuild = hdr->flags & ZVOL_OP_FLAG_REBUILD;
+	/*
+	 * Read metadata/IO# too in following cases:
+	 * 1. When Replica is helping other replica to rebuild
+	 * 2. When Replica is not healthy, need to return metadata too
+	 */
+	if (rebuild || ZVOL_IS_REBUILDED(zvol_state)) {
```

Comment:
if I understood correctly, if rebuild is true, we need metadata.. so, we need to go to 'else' case.
also, 'rebuild' is from IO hdr, can we make it as rebuildDataCmd or rebuildDataIO

Reply:
I'm assuming this happens only for reading data from replica where IO will be sent for rebuild to another one

Reply:
I have made changes as per suggestions.